pandas/io/formats/style_render.py

```diff
@@ -124,6 +133,7 @@ def __init__(
         self.hide_columns_: list = [False] * self.columns.nlevels
         self.hidden_rows: Sequence[int] = []  # sequence for specific hidden rows/cols
         self.hidden_columns: Sequence[int] = []
+        self.descriptors: list[Descriptor | tuple[str, Descriptor]] = []
```

Comment:
Why not make simplify this to `self.descriptors: list[tuple[str, Descriptor]]`?  Also, should it immutable, so that it would be `self.descriptors: tuple[tuple[str, Descriptor], ...]`?

Essentially if the user doesn't provide a name, then the name used internally, even if blank, is appended so that the tuple form is the only one that would ever need to be used.